airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -753,7 +753,7 @@ def cleanup(self, pod: k8s.V1Pod, remote_pod: k8s.V1Pod):
             self.process_pod_deletion(remote_pod, reraise=False)
 
             error_message = get_container_termination_message(remote_pod, self.base_container_name)
-            if self.skip_on_exit_code is not None:
+            if self.skip_on_exit_code:
```

Comment:
This will not work if the user want to skip on exit code 0

Reply:
Check operator constuctor, and read PR description again

Reply:
Well, I have to change few more things to make this work